pipeline_steps/common.groovy

```diff
@@ -1438,3 +1445,24 @@ Boolean isSkippable(String skip_pattern, String credentials) {
   }
   return skipIt
 }
+
+def findHookDir(String hook_dirs) {
+  String found_hook_dir = null
+  List split_hook_dirs = hook_dirs.split(/,\s?/)
+
+  for (Integer i = 0; i < split_hook_dirs.size(); i++) {
+    if (fileExists("${env.WORKSPACE}/${env.RE_JOB_REPO_NAME}/gating/${split_hook_dirs[i]}")) {
+      found_hook_dir = split_hook_dirs[i]
+      if (i > 0) {
+        print(
+          "DEPRECATION WARNING: This job is using a hook directory of `gating/${found_hook_dir}`.\n" \
+          + "DEPRECATION WARNING: This directory is being deprecated in favour of `gating/${split_hook_dirs[0]}`.\n" \
+          + "DEPRECATION WARNING: Please update your job accordingly!"
```

Comment:
Before we merge this, I'd like for us to agree on a timeframe for deprecation so we can add that to this warning message. I think it's pointless noting a deprecation unless we have a hard date for removal.

Reply:
I suspect we'll be the ones to update this so it probably doesn't matter too much. As long as the documentation is in order the teams will at least add new projects with the new names. We should put cards in the backlog to make sure it gets done.

Reply:
Set a removal date of 2018-08-24. I'll announce this change today.